integration-test-groups/cxf-soap/cxf-soap-client/src/main/java/org/apache/camel/quarkus/component/cxf/soap/client/it/CxfSoapClientRoutes.java

```diff
@@ -38,16 +47,45 @@ public class CxfSoapClientRoutes extends RouteBuilder {
     @ConfigProperty(name = "camel-quarkus.it.calculator.baseUri")
     String serviceBaseUri;
 
+    public static final String MESSAGE_RAW_SIMPLE_ADD = "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:ser=\"http://www.jboss.org/eap/quickstarts/wscalculator/Calculator\">\n"
+            +
+            "   <soapenv:Header/>\n" +
+            "   <soapenv:Body>\n" +
+            "      <ser:add>\n" +
+            "         <arg0>%s</arg0>\n" +
+            "         <arg1>%s</arg1>\n" +
+            "      </ser:add>\n" +
+            "   </soapenv:Body>\n" +
+            "</soapenv:Envelope>";
+
     @Override
     public void configure() {
 
         from("direct:simpleUriBean")
-                .to("cxf:bean:soapClientEndpoint?dataFormat=PAYLOAD");
+                .to("cxf:bean:soapClientEndpoint?dataFormat=POJO");
 
         from("direct:simpleUriAddress")
                 .to(String.format("cxf://%s?wsdlURL=%s&dataFormat=POJO&serviceClass=%s", calculatorServiceAddress(),
                         calculatorServiceWsdlUrl(), CalculatorService.class.getName()));
 
+        from("direct:simpleAddDataFormat")
+                .process(exchange -> {
+                    Map<String, Object> headers = exchange.getIn().getHeaders();
+                    String endpointDataFormat = headers.get("endpointDataFormat").toString();
+                    int[] numbers = exchange.getIn().getBody(int[].class);
+                    String xmlRequest = String.format(MESSAGE_RAW_SIMPLE_ADD, numbers[0], numbers[1]);
+                    if (DataFormat.RAW.name().equals(endpointDataFormat)) {
+                        exchange.getIn().setBody(xmlRequest);
+                    } else if (DataFormat.CXF_MESSAGE.name().equals(endpointDataFormat)) {
+                        try (InputStream is = new ByteArrayInputStream(xmlRequest.getBytes())) {
```

Comment:
The current machine's default encoding can be just anything.
```suggestion
                        try (InputStream is = new ByteArrayInputStream(xmlRequest.getBytes(StandardCharsets.UTF_8))) {
```

Reply:
Fixed